sdk/python/tests/compiler/testdata/loop_static.py

```diff
@@ -25,29 +25,61 @@ def empty(self):
 
 
 @dsl.pipeline(name='static-loop-pipeline')
-def pipeline(my_pipe_param='10'):
+def pipeline(my_pipe_param: str = '10'):
     loop_args = [1, 2, 3]
     with dsl.ParallelFor(loop_args) as item:
-        op1 = dsl.ContainerOp(
-            name="static-loop-inner-op1",
-            image="library/bash:4.4.23",
-            command=["sh", "-c"],
-            arguments=["echo op1 %s %s" % (item, my_pipe_param)],
-        )
-
-        op2 = dsl.ContainerOp(
-            name="static-loop-inner-op2",
-            image="library/bash:4.4.23",
-            command=["sh", "-c"],
-            arguments=["echo op2 %s" % item],
-        )
-
-    op_out = dsl.ContainerOp(
-        name="static-loop-out-op",
-        image="library/bash:4.4.23",
-        command=["sh", "-c"],
-        arguments=["echo %s" % my_pipe_param],
-    )
+        op1 = components.load_component_from_text("""
+        name: static-loop-inner-op1
+        description: static-loop-inner-op1
+        inputs:
+          - {name: input1, type: Integer}
+          - {name: input2, type: String}
+        implementation:
+          container:
+            image: library/bash:4.4.23
+            command:
+            - sh
+            - -c
+            args:
+            - |
```

Comment:
You don't need to run the `set -x` command here?

Reply:
original code doesn't have `set -x`. you want me to add `set -x` here?

Reply:
Oh i just wondering is the script still working without it, since I saw you added it in a lot of the examples in part 1 https://github.com/kubeflow/kfp-tekton/pull/685/files#diff-3b0c57e8f8c434d4d1e5338976ca3a0f5b41545519de957ffb53b84db38ba74eR59

Reply:
actually, the key part is `sh -c`. I don't add `set -e` for the script part now. want to eliminate the diff in the yaml files.

Reply:
cool thanks